### PR TITLE
chore: add docs for versioning and testing

### LINKED ISSUE #10: Document team norms around versioning and testing
To rebuild our foundation and question previous assumptions we should redo the ADRs that were all grouped together in a single ADR in the past here https://github.com/defenseunicorns/uds/blob/main/docs/adr/0006-uds-software-factory.md

Success Criteria:
- All ADRs found in the list above are documented here https://github.com/defenseunicorns/uds-software-factory/tree/main/adr
```[tasklist]
### Tasks
- [ ] SWF Testing
- [ ] SWF Versioning Scheme
```

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This adds docs for versioning and testing of SWF repos.

## Related Issue

Fixes #10

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-software-factory/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: docs/versioning.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# SWF Versioning
+
+> :warning: **Note:** Packages are the only artifacts published from the UDS Software Factory repositories, as bundles are meant to be _created_, not consumed, by users.
```
Comment (zachariahmiller) [inline docs/versioning.md:3]: @Racer159 Agreed with the intent but was operating under the assumption we would probably do something similar to uds-core as far as dev/demo bundles from this repo. Essentially a baseline for to test something on top of SWF that doesn involve having to build all the things just a `uds deploy uds-swf-dev:0.1.0` or for growth `uds deploy uds-swf-demo`

Comment (Racer159) [reply docs/versioning.md]: added some clarification on that point

**Review by naveensrinivasan — COMMENTED** [review]
Thank you!

A suggestion is to include diagrams.

``` mermaid
graph LR
    A[SWF Testing] --> B[Bundle Testing]
    A --> C[Package Testing]
    A --> D[Running Tests]
```
## Bundle Testing Details

``` mermaid
graph TD
    B[Bundle Testing] --> B1[Overall health of each package]
    B --> B2[Integration of packages with each other]
    B --> B3[Integration with uds-core]
    B --> B4[Integration with clusters beyond k3d]

 ```
## Package Testing
``` mermaid
graph TD
    C[Package Testing] --> C1[Health of the Package deployment]
    C --> C2[Package upgrades]
    C --> C3[Full matrix of package flavors]
    C --> C4[Integration with k3d-core-istio]
 ```

**Comment by Racer159** [comment]
> Thank you!
> 
> A suggestion is to include diagrams.
> 
> ```mermaid
> graph LR
>     A[SWF Testing] --> B[Bundle Testing]
>     A --> C[Package Testing]
>     A --> D[Running Tests]
> ```
> 
> ## Bundle Testing Details
> 
> ```mermaid
> graph TD
>     B[Bundle Testing] --> B1[Overall health of each package]
>     B --> B2[Integration of packages with each other]
>     B --> B3[Integration with uds-core]
>     B --> B4[Integration with clusters beyond k3d]
> 
> ```
> 
> ## Package Testing
> 
> ```mermaid
> graph TD
>     C[Package Testing] --> C1[Health of the Package deployment]
>     C --> C2[Package upgrades]
>     C --> C3[Full matrix of package flavors]
>     C --> C4[Integration with k3d-core-istio]
> ```

I think we _could_ but would want to know more about what they add (given for the most part these are flat lists).  Is the concern more visual distinction between them? (if so we could add emojis or italics/bold which IMO is easier to maintain / parse than mermaid)

**Comment by zachariahmiller** [comment]
> > Thank you!
> > A suggestion is to include diagrams.
> > 
> >   ```mermaid
> > graph LR
> >     A[SWF Testing] --> B[Bundle Testing]
> >     A --> C[Package Testing]
> >     A --> D[Running Tests]
> > ```
> >   
> >     
> >   
> >   
> > 
> >   
> > 
> >       graph LR
> >     A[SWF Testing] --> B[Bundle Testing]
> >     A --> C[Package Testing]
> >     A --> D[Running Tests]
> > 
> >     
> > 
> > ## Bundle Testing Details
> > 
> >   ```mermaid
> > graph TD
> >     B[Bundle Testing] --> B1[Overall health of each package]
> >     B --> B2[Integration of packages with each other]
> >     B --> B3[Integration with uds-core]
> >     B --> B4[Integration with clusters beyond k3d]
> > 
> > ```
> >   
> >     
> >   
> >   
> > 
> >   
> > 
> >       graph TD
> >     B[Bundle Testing] --> B1[Overall health of each package]
> >     B --> B2[Integration of packages with each other]
> >     B --> B3[Integration with uds-core]
> >     B --> B4[Integration with clusters beyond k3d]
> > 
> > 
> >     
> > 
> > ## Package Testing
> > 
> >   ```mermaid
> > graph TD
> >     C[Package Testing] --> C1[Health of the Package deployment]
> >     C --> C2[Package upgrades]
> >     C --> C3[Full matrix of package flavors]
> >     C --> C4[Integration with k3d-core-istio]
> > ```
> >   
> >     
> >   
> >   
> > 
> >   
> > 
> >       graph TD
> >     C[Package Testing] --> C1[Health of the Package deployment]
> >     C --> C2[Package upgrades]
> >     C --> C3[Full matrix of package flavors]
> >     C --> C4[Integration with k3d-core-istio]
> 
> I think we _could_ but would want to know more about what they add (given for the most part these are flat lists). Is the concern more visual distinction between them? (if so we could add emojis or italics/bold which IMO is easier to maintain / parse than mermaid)

I think the list is cleaner at this point. When we have do document a flow of the dependencies in the pipelines and between this and the package repos and how that correlates to testing and a release we can do diagramming, but in that case i would probably prefer something that is easier to read/prettier like a draw.io/lucid or whatever other tool that we include via a different means (image or link ref) rather than maintaining the mermaid in the md file.

**Review by zachariahmiller — APPROVED** [review]

**Pull request merged**